### ISSUE TITLE
Office Document Rendering Extremely Slow

### ISSUE THREAD

**Issue by ryester19** [nextcloud/android#11331]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. Open big spreadsheet file in Nextcloud Android app

### Expected behaviour

1. Spreadsheet should be rendered at about the same speed as in the web browser

### Actual behaviour

1. Spreadsheet rendering is much slower compared to opening the document in Fennec. It's borderline unusable because all UI actions are also severely delayed

**Nextcloud App:**

https://user-images.githubusercontent.com/17807258/215549725-7d7d7020-8803-4f07-9a83-305c1eb8f30d.mp4

Fennec Browser:

https://user-images.githubusercontent.com/17807258/215549901-8ee442a6-a8fb-4622-8407-48d5fbbfe5af.mp4

### Android version

Android 13

### Device brand and model

Oneplus 7 Pro

### Stock or custom OS?

Custom (explain in "additional information")

### Nextcloud android app version

20230124

### Nextcloud server version

25.0.3

### Using a reverse proxy?

Yes

### Android logs

[logs.txt](https://github.com/nextcloud/android/files/10538280/logs.txt)

### Server error logs

```bash
No relevant Nextcloud server output

There was some relevant log output from the Collabora server:

collabora[99363]: wsd-00001-00407 2023-01-30 17:38:06.427570 +0000 [ docbroker_00c ] WRN  ToClient-4d3: Tracker tileID 0:0:4388:10970:1097:1097:0 was dropped because of time out (5550ms). Tileprocessed message did not arrive in time.| wsd/ClientSession.cpp:2095
collabora[99363]: wsd-00001-00407 2023-01-30 17:38:06.427623 +0000 [ docbroker_00c ] WRN  ToClient-4d3: Tracker tileID 0:0:0:12067:1097:1097:0 was dropped because of time out (5550ms). Tileprocessed message did not arrive in time.| wsd/ClientSession.cpp:2095
collabora[99363]: wsd-00001-00407 2023-01-30 17:38:06.427678 +0000 [ docbroker_00c ] WRN  ToClient-4d3: Tracker tileID 0:0:1097:12067:1097:1097:0 was dropped because of time out (5550ms). Tileprocessed message did not arrive in time.| wsd/ClientSession.cpp:2095
collabora[99363]: wsd-00001-00407 2023-01-30 17:38:06.427732 +0000 [ docbroker_00c ] WRN  ToClient-4d3: Tracker tileID 0:0:2194:12067:1097:1097:0 was dropped because of time out (5550ms). Tileprocessed message did not arrive in time.| wsd/ClientSession.cpp:2095
collabora[99363]: wsd-00001-00407 2023-01-30 17:38:06.427784 +0000 [ docbroker_00c ] WRN  ToClient-4d3: Tracker tileID 0:0:3291:12067:1097:1097:0 was dropped because of time out (5550ms). Tileprocessed message did not arrive in time.| wsd/ClientSession.cpp:2095
collabora[99363]: wsd-00001-00407 2023-01-30 17:38:06.427837 +0000 [ docbroker_00c ] WRN  ToClient-4d3: Tracker tileID 0:0:4388:12067:1097:1097:0 was dropped because of time out (5550ms). Tileprocessed message did not arrive in time.| wsd/ClientSession.cpp:2095
```

### Additional information

Nextcloud Office version 7.1.0. Coolwsd version 22.05.9.3

Using custom Android ROM LineageOS 20.0

The test spreadsheet being used is 2255 rows long.

Not sure if I should have opened this issue in the android, richdocument, or collabora code repo. I was not able to reproduce the issue anywhere else except loading through the Android app.

**Comment by AlvaroBrey** [comment]
The app loads it through a system WebView. Can you check if you have any updates pending for that? https://play.google.com/store/apps/details?id=com.google.android.webview&hl=en&gl=US

**Comment by outerSpaceToast** [comment]
I can confirm this bug. Exactly same behaviour here. Slow in Nextcloud app, fast in Browser

Android version
Android 13, OneUI 5.1

Device brand and model
Samsung Tab S8

Stock or custom OS?
Stock

Nextcloud android app version
3.24.1

Nextcloud server version
26.0.1

Using a reverse proxy?
No

**Comment by ryester19** [comment]
Interesting. I kinda let this drop because I thought it was some problem with LineageOS and what they might have done with their Webview builds, but a stock device having this problem makes me think otherwise now

@AlvaroBrey can we re-open this?

**Comment by ryester19** [comment]
At least for me, the problem is pretty much solved after the Collabora CODE 1.27.2 release
